sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/V2WriteAnalysisSuite.scala

```diff
@@ -744,6 +744,46 @@ abstract class V2WriteAnalysisSuiteBase extends AnalysisTest {
     }
   }
 
+  test ("SPARK-44414: Fixed matching check for CharType/VarcharType") {
+    // check varchar type
+    val json1 = "{\"__CHAR_VARCHAR_TYPE_STRING\":\"varchar(80)\"}"
```

Comment:
Consider using `Seq("{\"__CHAR_VARCHAR_TYPE_STRING\":\"char(80)\"}", "{\"__CHAR_VARCHAR_TYPE_STRING\":\"varchar(80)\"}").foreach { ... ` to minimize duplicates?

Reply:
> Consider using `Seq("{\"__CHAR_VARCHAR_TYPE_STRING\":\"char(80)\"}", "{\"__CHAR_VARCHAR_TYPE_STRING\":\"varchar(80)\"}").foreach { ... ` to minimize duplicates?

updated. Thanks for reviewing @yaooqinn

Reply:
Do we also have an e2e test case to add to CharVarcharTestSuite?